Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/checkins/checkout/ContactJournalEntryCreator.kt

```diff
@@ -0,0 +1,76 @@
+package de.rki.coronawarnapp.presencetracing.checkins.checkout
+
+import dagger.Reusable
+import de.rki.coronawarnapp.contactdiary.model.ContactDiaryLocation
+import de.rki.coronawarnapp.contactdiary.model.ContactDiaryLocationVisit
+import de.rki.coronawarnapp.contactdiary.model.DefaultContactDiaryLocation
+import de.rki.coronawarnapp.contactdiary.model.DefaultContactDiaryLocationVisit
+import de.rki.coronawarnapp.contactdiary.storage.repo.ContactDiaryRepository
+import de.rki.coronawarnapp.eventregistration.checkins.CheckIn
+import de.rki.coronawarnapp.eventregistration.checkins.split.splitByMidnightUTC
+import de.rki.coronawarnapp.util.TimeAndDateExtensions.toLocalDateUtc
+import kotlinx.coroutines.flow.first
+import org.joda.time.Duration
+import org.joda.time.Seconds
+import timber.log.Timber
+import javax.inject.Inject
+import kotlin.math.roundToLong
+
+@Reusable
+class ContactJournalEntryCreator @Inject constructor(
+    private val diaryRepository: ContactDiaryRepository
+) {
+
+    suspend fun createEntry(checkIn: CheckIn) {
+        if (checkIn.createJournalEntry) {
```

Comment:
I'd prefer that the `if (checkIn.createJournalEntry)' would happen in `CheckOutHandler`
or
Renaming this `tryCreateEntry` or `maybeCreateEntry` and then using an early `return` if `!createJournalEntry`, but the first approach seems cleaner.

Reply:
will be handled in a follow up PR